mlir/unittests/Analysis/Presburger/IntegerRelationTest.cpp

```diff
@@ -207,3 +207,86 @@ TEST(IntegerRelationTest, swapVar) {
   EXPECT_TRUE(swappedSpace.getId(VarKind::Symbol, 1)
                   .isEqual(space.getId(VarKind::Domain, 1)));
 }
+
+TEST(IntegerRelationTest, mergeAndAlignSymbols) {
+  IntegerRelation rel =
+      parseRelationFromSet("(x, y, z, a, b, c)[N, Q] : (a - x - y == 0, "
+                           "x >= 0, N - b >= 0, y >= 0, Q - y >= 0)",
+                           3);
```

Comment:
Can you add two tests:

- Unrelated identifiers: [A, B, C, D], [E, F, G, H]
- Unrelated prefix, then related identifiers: [A, B, C, D], [E, F, C, D]